src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -617,19 +666,41 @@ impl SimulationElement for A320EngineDrivenPumpController {
             panic!("The A320 only supports two engines.");
         }
     }
+
+    fn receive_power(&mut self, supplied_power: &SuppliedPower) {
+        if self.engine_number == 1 {
+            self.is_powered = supplied_power
+                .potential_of(&self.powering_buses[0])
+                .is_powered();
+        } else if self.engine_number == 2 {
+            self.is_powered = supplied_power
+                .potential_of(&self.powering_buses[0])
+                .is_powered()
+                || supplied_power
+                    .potential_of(&self.powering_buses[1])
+                    .is_powered();
+        } else {
+            panic!("The A320 only supports two engines.");
+        }
```

Comment:
Just hit the button:

```suggestion
        self.is_powered = self
            .powering_buses
            .iter()
            .any(|bus| supplied_power.potential_of(&bus).is_powered());
```

Reply:
Beautiful :)